tests/conftest.py

```diff
@@ -119,11 +118,31 @@ def generate_tests(self, smoke_test_data, random_test_data=[]):
         _: list
             Tests.
         """
-        smoke_tests = [
-            pytest.param(*data.values(), marks=smoke) for data in smoke_test_data
-        ]
-        random_tests = [pytest.param(*data, marks=random) for data in random_test_data]
-        return smoke_tests + random_tests
+        tests = []
+        if smoke_test_data:
+            smoke_tests = [
+                pytest.param(*data.values(), marks=pytest.mark.smoke)
+                for data in smoke_test_data
+            ]
+            tests += smoke_tests
+        if random_test_data:
+            random_tests = [
+                pytest.param(*data, marks=pytest.mark.random)
+                for data in random_test_data
+            ]
+            tests += random_tests
+        return tests
+
+    def log_exp_composition(self, space, num_samples=10, max_n=10, num_n=5, **kwargs):
```

Comment:
I understand that you chose `num` to avoid having `n_n`, but I think we should still use `n_samples` and `n_n`, to follow the conventions in the code base.

Reply:
well I think it's bad convention then.

Reply:
we should try to change that

Reply:
change `n_` to `num_`? For sure, if you want to change it in the whole database, I"m open to it! (I'm just too lazy to do it). Still, we chose `n_` because it was shorter.

Reply:
For the time being we should either move to `num_` in the whole code base (consistency of the codebase is the most important thing), or use `n_` for now. I let you choose!

Reply:
I will chose `n_` for now and update it in separate PR